### ISSUE TITLE
Array.map error after upgrading packages and changing to pnpm

### ISSUE THREAD

**Issue by azehintense** [mauriciolauffer/wdio-qunit-service#24]
Hi. I can't get version 1.0.0 to work. What am i doing wrong? The packages seem to work otherwise.

I get this error:
![image](https://github.com/user-attachments/assets/732ccf6a-07d1-4cf7-b073-1fab855729e9)

This is my package.json:
```
{
    "name": "cdt",
    "version": "2.0.0",
    "private": true,
    "packageManager": "pnpm@9.12.2",
    "engines": {
        "node": ">=18.14.0",
        "pnpm": ">=9"
    },
    "devDependencies": {
        "@openui5/types": "^1.130.0",
        "@types/node": "^22.9.0",
        "@types/qunit": "^2.19.12",
        "@ui5/middleware-code-coverage": "^1.1.1",
        "@wdio/browser-runner": "^9.2.11",
        "@wdio/cli": "^9.2.11",
        "@wdio/devtools-service": "^8.40.2",
        "@wdio/junit-reporter": "^9.2.2",
        "@wdio/local-runner": "^9.2.11",
        "@wdio/mocha-framework": "^9.2.8",
        "@wdio/spec-reporter": "^9.2.8",
        "@wdio/static-server-service": "^9.2.2",
        "@wdio/types": "^9.2.2",
        "@wdio/visual-service": "^4.1.3",
        "chart.js": "^3.9.1",
        "concurrently": "^8.2.2",
        "eslint": "^9.14.0",
        "eslint-config-mlauffer-nodejs": "^3.0.0",
        "eslint-plugin-mocha": "^10.5.0",
        "eslint-plugin-wdio": "^9.2.11",
        "file-system": "^1.2.9",
        "globals": "^15.12.0",
        "prettier": "^3.3.3",
        "qunit": "^2.22.0",
        "ts-node": "^10.9.2",
        "tsx": "^4.19.2",
        "typescript": "^5.6.3",
        "typescript-eslint": "^8.13.0",
        "ui5-tooling-less": "^3.2.1",
        "ui5-tooling-modules": "^3.18.4",
        "vitest": "^2.1.4",
        "wait-on": "^7.2.0",
        "wdio-qunit-service": "^1.0.0"
    },
    "scripts": {
        "preinstall": "npx only-allow pnpm",
        "installAndUpdate": "pnpm install && pnpm update && pnpm audit fix --force || true",
        "test": "TYPE=unit pnpm run runTests",
        "runTests": "npx concurrently -n \"server,tests\" -k -t 60s -s \"command-tests\" \"ui5 serve --p=8484\" \"npx wait-on http://localhost:8484 && pnpm run $TYPE\"",
        "unit": "npx wdio run ./webapp/test/unit/wdio.conf.js --remote-debugging-port=9222 --remote-debugging-host=0.0.0.0 --debug",
        "silentServe": "ui5 serve --p=8484",
        "updateWDIO": "rm -rf node-modules package-lock.json && pnpm i @wdio/cli@latest @wdio/local-runner@latest @wdio/mocha-framework@latest --save-dev",
        "integrationTest": "TYPE=integration pnpm run runTests --verbose",
        "integration": "npx wdio run ./webapp/test/integration/wdio.conf.js --remote-debugging-port=9222 --remote-debugging-host=0.0.0.0",
        "serve": "ui5 serve --open index.html --p=8081",
        "build": "ui5 build --all"
    },
    "dependencies": {
        "@sap-theming/theming-base-content": "^11.22.0",
        "@ui5/cli": "^3.11.4",
        "less": "^4.2.0",
        "selenium-webdriver": "^4.26.0"
    }
}

```

And this is my new pnpm-workspace.yaml:
```
packages:
  # include packages in subfolders (e.g. apps/ and packages/)
  - "apps/**"
  - 'packages/**'
  # if required, exclude some directories
  - '!**/test/**'
```

Btw love your work!

**Comment by mauriciolauffer** [comment]
hi @azehintense , thanks for catching this one  :)

The error was due to array values in QUnit assertion results `actual/expected` as seem in this [SAP openui5-sample-app](https://github.com/SAP/openui5-sample-app/blob/3ebd1421fb4770c88718b2f0276a8f679f668d50/webapp/test/unit/controller/App.controller.js#L33-L41) example.


I've fixed the issue in [v1.0.1](https://github.com/mauriciolauffer/wdio-qunit-service/releases/tag/v1.0.1) and added some tests for these scenarios.
https://github.com/mauriciolauffer/wdio-qunit-service/blob/de3da8ea42a16839cd285cd988d2f41688815f0a/examples/fixtures/qunit-no-modules-sync.js#L9-L22

**Comment by azehintense** [comment]
Thank you <3
